Makefile

```diff
@@ -0,0 +1,101 @@
+GOOS:=linux
+VERSION:=$(shell git describe --abbrev=4 --always --tags)
+NAME:=mesos-dns
+CONTAINERNAME:=$(NAME)
+CIRCLE_TEST_REPORTS?=$(PWD)
+CIRCLE_ARTIFACTS?=$(PWD)/build
+COVERALLS_REPO_TOKEN?="tokennotset"
+DOCKER?=
+
+all: $(if $(DOCKER), dockerbuild) test build
+
+deps:
+	go get -u github.com/kardianos/govendor
+	go get -u github.com/mitchellh/gox
+	go get -u github.com/alecthomas/gometalinter
+	go get -u github.com/axw/gocov/gocov
+	go get -u github.com/mattn/goveralls
+	go get -u github.com/jstemmer/go-junit-report
+	gometalinter --install
+	govendor sync
+	go install ./...
+	go test -i ./...
+
+dockerbuild:
+	docker build -t $(CONTAINERNAME)-build -f Dockerfile.test .
+
+test: fmttest metalinter gocov junit gorace
+
+## Ignore vendored dependencies
+## Return 1 if any files found that haven't been formatted
+fmttest: $(if $(DOCKER), dockerbuild)
+	$(if $(DOCKER), docker run --rm $(CONTAINERNAME)-build) gofmt -l . | \
+								awk '!/^vendor/ { print $0; err=1 }; END{ exit err }'
+
+junit: $(if $(DOCKER), dockerbuild)
+	mkdir -p $(CIRCLE_TEST_REPORTS)/junit
+	@# This seems odd, but if you attach stdin without attaching to an
+	@# output ( stdout or stderr ), the container runs in the background
+	$(if $(DOCKER), docker run --rm $(CONTAINERNAME)-build) go test -v -timeout=10m ./... | \
+	$(if $(DOCKER), docker run --rm -i -a stdin -a stdout -a stderr $(CONTAINERNAME)-build ) go-junit-report > $(CIRCLE_TEST_REPORTS)/junit/alltests.xml
```

Comment:
We could try, but I'd be concerned about not cleaning up when people are running it locally.